WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsCollectionViewFlowLayout.swift

```diff
@@ -0,0 +1,204 @@
+import UIKit
+
+class ReaderInterestsCollectionViewFlowLayout: UICollectionViewFlowLayout {
+    @IBInspectable public var itemSpacing: CGFloat = 6
+    @IBInspectable public var cellHeight: CGFloat = 40
+
+    private var layoutAttributes: [UICollectionViewLayoutAttributes] = []
+    private var layoutDirection: UIUserInterfaceLayoutDirection {
+        return collectionView?.effectiveUserInterfaceLayoutDirection ?? .leftToRight
+    }
+
+    // The content width minus the content insets used when calculating rows, and centering
+    private var maxContentWidth: CGFloat {
+        guard let collectionView = collectionView else {
+            return 0
+        }
+
+        let contentInsets: UIEdgeInsets = collectionView.contentInset
+
+        return collectionView.bounds.width - (contentInsets.left + contentInsets.right)
+    }
+
+    /// The calculated content size for the view
+    private var contentSize: CGSize = .zero
+
+    override open var collectionViewContentSize: CGSize {
+        return contentSize
+    }
+
+    override func prepare() {
+        guard let collectionView = collectionView else {
+            return
+        }
+
+        let contentInsets = collectionView.contentInset
+        let isRTL = layoutDirection == .rightToLeft
+
+        // The current row used to calculate the y position and the total content height
+        var currentRow: CGFloat = 0
+
+        // Keeps track of the previous items frame so we can properly calculate the current item's x position
+        var previousFrame: CGRect = .zero
+
+        let numberOfItems: Int = collectionView.numberOfItems(inSection: 0)
+
+        for item in 0 ..< numberOfItems {
+            let indexPath: IndexPath = IndexPath(row: item, section: 0)
+
+            let itemSize = sizeForItem(at: indexPath)
+            var frame: CGRect = CGRect(origin: .zero, size: itemSize)
+
+            if item == 0 {
+                let minX: CGFloat = isRTL ? maxContentWidth - frame.width : 0
+                frame.origin = CGPoint(x: minX, y: contentInsets.top)
+            } else {
+                if isRTL {
```

Comment:
Very nitpicky but I'd personally do:
`frame.origin.x = isRTL ? previousFrame.minX - itemSpacing - frame.width : previousFrame.maxX + itemSpacing` here since there are multiple levels of if else and since you're using Ternary for other cases in this method

Reply:
Oh I like that more. 🙇‍♀️